lib/service/system.rb

```diff
@@ -72,9 +72,21 @@ def path
       root? ? boot_path : user_path
     end
 
-    def domain_target
+    def domain_target_needs_background?(service)
+      plist_data = service.generate_plist(nil)
+      plist = begin
+        Plist.parse_xml(plist_data)
+      rescue
```

Comment:
A comment here explaining why it's `nil` would be nice.

Reply:
This is not resolved yet.

Reply:
I did add [a comment explaining that](https://github.com/Homebrew/homebrew-services/pull/512/files#diff-628d1358d1f6da549a3320bbf39688c4f7800fb6e7c0143d252549c7f39c928f), I think it's just barely not shown here because you tied your comment to line `plist_data = ...` line and not `def domain_target_needs_background?(service)`. :>

Unless you meant why `nil` under `rescue`? In that case: isn't it already pretty clear that it indicates a parsing error and we shouldn't/can't even check the particular key?

Reply:
> Unless you meant why nil under rescue? In that case: isn't it already pretty clear that it indicates a parsing error and we shouldn't/can't even check the particular key?

Yes, a comment directly above the `nil` would be preferred. Ideally, also catch a more specific exception.

Reply:
Alright, I think that should about cover it.